File: mykoob/models/school.py
from .school_class import SchoolClass
from mykoob.utils import show

class School:
    def __init__(self, data: dict):
        self.region = data['region']
        self.country_code = data['country_code']
        self.id = data['school_id']
        self.name = data['name']
        self.user_id = data['school_user_id']
        self.gate_enabled = data['school_gate_enabled']
        self.school_classes: list[SchoolClass] = [SchoolClass(school_class) for school_class in data['class']]
        
        show("School was initialized successfully")
        
    def __repr__(self) -> str:
        return f"School(id={self.id} name={self.name})"

File: javascript/src/models/school.js
import { SchoolClass } from './schoolClass.js';
import { show } from '../utils.js';

class School {
    /**
     * Initializes the School object.
     * @param {Object} data - The data representing the school.
     */
    constructor(data) {
        this.region = data.region;
        this.countryCode = data.country_code;
        this.id = data.school_id;
        this.name = data.name;
        this.userId = data.school_user_id;
        this.gateEnabled = data.school_gate_enabled;
        this.schoolClasses = data.class.map(schoolClass => new SchoolClass(schoolClass));
    }
    /**
     * Returns a string representation of the School object.
     * @returns {string} - String representation of the school.
     */
    toString() {
        return `School(id=${this.id}, name=${this.name})`;
    }
}

export { School };
